src/network/knownnodes.py

```diff
@@ -95,12 +99,44 @@ def saveKnownNodes(dirName=None):
 
 
 def addKnownNode(stream, peer, lastseen=None, is_self=False):
-    """Add a new node to the dict"""
+    """
+    Add a new node to the dict or update lastseen if it already exists.
+    Do it for each stream number if *stream* is `Iterable`.
+    Returns True if added a new node.
+    """
+    # pylint: disable=too-many-branches
+    if isinstance(stream, Iterable):
+        with knownNodesLock:
```

Comment:
Since now we're now using `knownNodesLock` in a reentrant fashion, we should redefine it as a `threading.RLock` to avoid deadlocks.

Reply:
Done. Unfortunately threads is not my thing.